src/dotnet/BuildServer/RazorPidFile.cs

```diff
@@ -33,32 +33,46 @@ public static RazorPidFile Read(FilePath path, IFileSystem fileSystem = null)
         {
             fileSystem = fileSystem ?? FileSystemWrapper.Default;
 
-            using (var stream = fileSystem.File.OpenRead(path.Value))
-            using (var reader = new StreamReader(stream, Encoding.UTF8))
+            try
             {
-                if (!int.TryParse(reader.ReadLine(), out var processId))
+                using (var stream = fileSystem.File.OpenFile(
+                    path.Value,
+                    FileMode.Open,
+                    FileAccess.Read,
+                    FileShare.Write | FileShare.Delete,
+                    4096,
+                    FileOptions.None))
+                using (var reader = new StreamReader(stream, Encoding.UTF8))
                 {
-                    return null;
-                }
+                    if (!int.TryParse(reader.ReadLine(), out var processId))
+                    {
+                        return null;
+                    }
 
-                if (reader.ReadLine() != RazorServerType)
-                {
-                    return null;
-                }
+                    if (reader.ReadLine() != RazorServerType)
+                    {
+                        return null;
+                    }
 
-                var serverPath = reader.ReadLine();
-                if (string.IsNullOrEmpty(serverPath))
-                {
-                    return null;
-                }
+                    var serverPath = reader.ReadLine();
+                    if (string.IsNullOrEmpty(serverPath))
+                    {
+                        return null;
+                    }
 
-                var pipeName = reader.ReadLine();
-                if (string.IsNullOrEmpty(pipeName))
-                {
-                    return null;
-                }
+                    var pipeName = reader.ReadLine();
+                    if (string.IsNullOrEmpty(pipeName))
+                    {
+                        return null;
+                    }
 
-                return new RazorPidFile(path, processId, new FilePath(serverPath), pipeName);
+                    return new RazorPidFile(path, processId, new FilePath(serverPath), pipeName);
+                }
+            }
+            catch (Exception ex) when (ex is IOException || ex is UnauthorizedAccessException)
+            {
+                // Any failure to read the file will ignore the file
```

Comment:
I think maybe the `BuildServerProvider` should be given an `IReporter` it can use to report a warning if it failed to open a pid file?  I don't like silent failures.

Reply:
I've decided I'm going to change this and add a test that fails simulates a failure here.